port/linux/omrvmem.c

```diff
@@ -869,6 +877,17 @@ reserveLargePages(struct OMRPortLibrary *portLibrary, struct J9PortVmemIdentifie
 	return memoryPointer;
 }
 
+uintptr_t
+omrvmem_advise_hugepage(struct OMRPortLibrary *portLibrary, void* address, uintptr_t byteAmount)
```

Comment:
I am not sure about this new API. This forces all users of VMem to specifically call this API at exactly the right time. I think this should happen in the reserve / commit call directly based on the same or similar decision.

Reply:
This api should be called after `mmap()` call succeeded, given that this API is only intended for use under `[madvise]` setting, I am not sure we should force all vmem allocation to be flagged for THP. As enabling/disabling this option will make no difference from setting `THP => always/never` on the user side. 

For prototyping, if I change this to an internal function that is triggered after a successful `mmap()` call, would that be better?

Reply:
yes I believe it would be

Reply:
I added a call to the API after `getMemoryInRangeForDefaultPages` successfully completed to set hugepage flag after `mmap()` and `mprotect()` succeeded.